packages/hardhat-plugin/src/utils/verifyEtherscanContract.ts

```diff
@@ -0,0 +1,38 @@
+import type { Etherscan } from "@nomicfoundation/hardhat-verify/etherscan";
+import type { VerifyInfo } from "@nomicfoundation/ignition-core";
+
+export async function verifyEtherscanContract(
+  etherscanInstance: Etherscan,
+  { address, compilerVersion, sourceCode, name, args }: VerifyInfo
+): Promise<
+  { type: "success"; contractURL: string } | { type: "failure"; reason: Error }
+> {
+  try {
+    const { message: guid } = await etherscanInstance.verify(
+      address,
+      sourceCode,
+      name,
+      compilerVersion,
+      args
+    );
+
+    await new Promise((res) => setTimeout(res, 1000));
```

Comment:
Do we need to retry up to a certain timeout?

Reply:
Looking at the implementation in HH-viem, it already polls. @schaable is the 1s timeout necessary or can we call `getVerificationStatus` immediately?

Reply:
It shouldn't be necessary. `verify` returns immediately, and `getVerificationStatus` does the polling.